### PR TITLE
Refactor user settings selectors to TS and remove lodash

### PR DESCRIPTION
**Pull request by daledupreez**
#### Changes proposed in this Pull Request

* This PR refactors the `getUnsavedUserSettings()`, `getUserSetting()`, and `getUserSettings()` selectors to be written in TypeScript, and removes their use of the `get()` function from `lodash`

#### Testing instructions

* It will be pretty difficult to test this directly, but it may be easier to make sure that `/me/account` continues to work as expected when updating various values.

Related to #61734 , which added the types at a higher level (and not in the right place)

### CODE REVIEW

File: client/state/selectors/get-user-setting.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import 'calypso/state/user-settings/init';
+
+export type UserSettingValue = boolean | number | string;
+
+/**
+ * Given a settingName, returns that setting if it exists or null
+ *
+ * @param  {AppState} state Global state tree
+ * @param  {string} settingName - setting name
+ * @returns {boolean | number | string | null} setting value
```
Comment (olaseni) [inline client/state/selectors/get-user-setting.ts]: We could use this type in the comments i.e. `UserSettingValue`

Comment (AllTerrainDeveloper) [reply client/state/selectors/get-user-setting.ts]: We can even remove completely the docblock as it's now typed

File: client/state/selectors/get-user-settings.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+export type UserSettingsType = Record< string, UserSettingValue >;
+
+/**
+ * Returns all user settings as one object
+ *
+ *
+ * @param {AppState} state Global state tree
+ * @returns {UserSettingsType | null} dictionary with the setting names and values
+ */
```
Comment (AllTerrainDeveloper) [inline client/state/selectors/get-user-settings.ts]: Same thing, I would say that we can remove it and still self descriptive enough, isn't it?

File: client/state/selectors/get-unsaved-user-settings.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import 'calypso/state/user-settings/init';
+
+/**
+ * Returns all unsaved user settings as one object
+ *
+ *
+ * @param {AppState} state Global state tree
+ * @returns {UserSettingsType | null} dictionary with the setting names and values
+ */
```
Comment (AllTerrainDeveloper) [inline client/state/selectors/get-unsaved-user-settings.ts]: Same remark as below

**Review by AllTerrainDeveloper — APPROVED** [review]
LGTM! And /me/account still working on the Calypso live link (no EOT screenshot due to too much information to hide 😪 )

**Pull request merged**